CMakeLists.txt

```diff
@@ -96,7 +98,6 @@ if(OPENMP_FOUND AND BUILD_WITH_MULTITHREADS)
   ADD_DEFINITIONS(-DWITH_MULTITHREADING)
   ADD_DEFINITIONS(-DWITH_NTHREADS=${BUILD_WITH_NTHREADS})
   SET(OMP_NUM_THREADS ${BUILD_WITH_NTHREADS})
```

Comment:
Why are you removing this compilation flag?

If I correct, we need to pass `WITH_NTHREADS` to our code, last time a similar modification broke the code.

However, and generally speaking, you should follow the rule of thumb "If it ain't broke, don't fix it".

Please avoid to push more stuff in this branch. Otherwise, I need to repeat the review of really big PR.

Reply:
You can still keep working in other branches from this point, this practice helps
 - to avoid introducing bugs in the code,
 - to have a higher quality code review.

Reply:
Hi Carlos,
> However, and generally speaking, you should follow the rule of thumb "If it ain't broke, don't fix it".

Could you please point to the un-broken part that I fixed, and I can be more precise in explaining what was broken? It is a bit hard, since the commit messages already explain in brief what the changes are about.

Reply:
> Why are you removing this compilation flag?

I need to make sure that the repo works for the downstream people as well. For that, and after discussion with @nim65s, we decided to put the preprocessor macro definitions inside crocoddylConfig.cmake, instead of pkg-config `.pc` file.

It looks like the line `LIST(APPEND CFLAGS_DEPENDENCIES ...)` was taken from https://github.com/stack-of-tasks/pinocchio/blob/master/CMakeLists.txt#L101. In pinocchio, this list was being used for updating the pkg-config `.pc` file. This is being done at https://github.com/stack-of-tasks/pinocchio/blob/master/CMakeLists.txt#L293. However, since I'm adding the macro definition inside the `crocoddylConfig.cmake`, we don't need this list here. 

Moreover, the list  `CFLAGS_DEPENDENCIES` is _not defined anywhere_ in the current CMakeLists.txt. And after this undefined list is amended in this line, it is _not used anywhere_ in the current CMakeLists.txt. If there was a compiler that compiled CMakeLists, it would give you an error that this variable is undefined, and once you define it, it would give you a warning that this variable is defined but not used. Does this answer your question?

Reply:
Keep .pc file compatibility. Some project still relies on it. Do not breack downstream users.

Reply:
Any downstream user who uses a .pc file _would not_ see any break in compatibility. This PR creates some compatibility with cmake for people using crocoddyl.

Reply:
Just to mention that this macro comes along with the target crocoddyl. You don't need to export them in this way. This is the reason to use PRIVATE and PUBLIC keywords.

Reply:
On second thought, I think it might be better if we don't introduce this macro in the header files _at all_. My goal is to make the codegen headers available to any downstream repo, regardless of how they compiled crocoddyl. However, currently, and also in the solution I proposed, it is being hidden under the `#ifdef ... #endif` structure, which means any dependency would have to recompile crocoddyl, just to have access to these macros. Which should not be the case. The headers should always remain available.

What I propose is to 
1) remove the macros `CROCODDYL_WITH_CODEGEN` from `include/crocoddyl/core/codegen/action-base.hpp` and `include/crocoddyl/core/fwd.hpp`. 

2) I understand that the `PINOCCHIO_WITH_CODEGEN` macro is also being removed from pinocchio. However, untill this becomes available in the next release, we still need to make sure that turning on codegen inside crocoddyl means he has access to pinocchio functionality, regardless of pinocchio compilation setting. So let's define these macros (`PINOCCHIO_WITH_CODEGEN ...`) when we turn on codegen in crocoddyl. After pinocchio removes these macros, we can too.

Reply:
> * remove the macros `CROCODDYL_WITH_CODEGEN` from `include/crocoddyl/core/codegen/action-base.hpp` and `include/crocoddyl/core/fwd.hpp`.
> 
> * I understand that the `PINOCCHIO_WITH_CODEGEN` macro is also being removed from pinocchio. However, untill this becomes available in the next release, we still need to make sure that turning on codegen inside crocoddyl means he has access to pinocchio functionality, regardless of pinocchio compilation setting. So let's define these macros (`PINOCCHIO_WITH_CODEGEN ...`) when we turn on codegen in crocoddyl. After pinocchio removes these macros, we can too.


If I understood correctly, you propose to use `PINOCCHIO_WITH_CODEGEN` inside the `crocoddyl/core/fwd.hpp` file. If not, how does the code handle the forward declaration?

I will take the opportunity to ask you the follows 
 - "Does our cmake code detect the Pinocchio macros for compiling or not codegen related code?"
 - "If so, what happens for other cases such as CasADi"?



> On second thought, I think it might be better if we don't introduce this macro in the header files _at all_.


We also have the `NTHREADS` macro. Is this also a problem for codegen? If so, I don't see how we could remove it, since the number of threads has to be defined at compile time.